core/src/main/java/org/infinispan/util/concurrent/locks/impl/StripedLockContainer.java

```diff
@@ -34,13 +34,10 @@ public StripedLockContainer(int concurrencyLevel) {
    }
 
    @Inject
-   void inject(@ComponentName(KnownComponentNames.BLOCKING_EXECUTOR) Executor executor, TimeService timeService) {
+   void inject(@ComponentName(KnownComponentNames.NON_BLOCKING_EXECUTOR) Executor nonBlockingExecutor,
+         TimeService timeService) {
       for (int i = 0; i < sharedLocks.length; i++) {
-         if (sharedLocks[i] == null) {
-            sharedLocks[i] = new InfinispanLock(executor, timeService);
-         } else {
-            sharedLocks[i].setTimeService(timeService);
-         }
+         sharedLocks[i] = new InfinispanLock(nonBlockingExecutor, timeService);
```

Comment:
I'm starting to think figuring out why a lock isn't unlocked after a `BasicComponentRegistry.rewire()` is too hard, and tests almost never replace executors anyway, so the master version of the method is safer.